src/CMakeLists.txt

```diff
@@ -132,24 +128,24 @@ else(WITH_CUDA)
     set_target_properties(CCTag PROPERTIES VERSION ${PROJECT_VERSION})
   endif( COMPILE_STATIC_CCTAG_LIBRARY )
 
-  target_include_directories(CCTag 
+endif(WITH_CUDA)
```

Comment:
nope, you cannot do that, "share the code", as CUDA does not yet support the PUBLIC/PRIVATE stuff yet, it uses the old-fashion include_directories etc.